cylc/flow/data_store_mgr.py

```diff
@@ -854,7 +872,14 @@ def add_pool_node(self, name, point):
         ).id
         self.all_task_pool.add(tp_id)
 
-    def generate_ghost_task(self, tp_id, itask, is_parent=False):
+    def generate_ghost_task(
+        self,
+        tokens,
+        point,
+        flow_nums,
+        is_parent=False,
+        itask=None
+    ):
         """Create task-point element populated with static data.
```

Comment:
No, it's not `TaskProxy` it is the data-store task proxy node (`PBTaskProxy`)..

`TaskProxy` object is used for alot of the attributes to fill out this protobuf class i.e.
https://github.com/cylc/cylc-flow/blob/259956bb60e093b23aa1f6f8e0a7a545e01d2052/cylc/flow/data_store_mgr.py#L1162

and others.. It's also convenient that the pool is made up of TaskProxy, so that can be passed in to gather the info from.